### DISCUSSION TITLE
name 'take' is not defined with --pre version

### DISCUSSION THREAD

**Discussion by gthvn1** [hylang/hy#2304] [Q&A] [answered]
Hi,

 It is my first message so I'd like to start by saying thanks for "Hy". I"m using python but I love LISP.  So thank you.

 I looked the documentation and when I open the alpha version I didn't find _built-ins_ and "_Hy core_" chapters that I can find in stable. Also when I trying simple things like` (take 10 (range 20))` with the alpha version I got "take" is not defined. With stable "take" is defined.

 Should I import something with alpha version? Is it a really different version that is not compatible?

Thanks for your help and I hope that I will able to contribute. I love the idea of LISP & Python :)

**Comment by scauligi** [comment, marked as answer]
We moved a lot of the non-core stuff out of the `hy` module to keep it relatively lightweight and easier for us to manage.
You can import `take` and related functions from Toolz. Also if you're using macros, most have been split off into a separate module called `hyrule`.

If you're trying to find a specific function/macro that seems missing, you can probably find where it went by looking through [our NEWS.rst file](https://github.com/hylang/hy/blob/master/NEWS.rst)!
